Extend/Controller/OrderController.php

```diff
@@ -286,4 +377,118 @@ private function _isFinalizeOrderSuccessful($iSuccess)
+        $oPaymentGateway = oxNew(PaymentGateway::class);
+
+        /**
+         * @var $oBasket Basket
+         */
+        $oBasket = $this->getBasket();
+
+        /**
+         * @var $oOrder Order
+         */
+        $oOrder = oxNew(Order::class);
+        $oOrder->createTemp($oBasket, $this->getUser());
+
+        $oTransaction = $oPaymentGateway->createTransaction($oBasket, $oOrder);
+
+        $oTransaction->setAmount(new Amount(
+            $oBasket->getPrice()->getBruttoPrice(),
+            $oBasket->getBasketCurrency()->name
+        ));
+        $oTransaction->setConfig($this->_getCreditCardPaymentMethodConfig()->get(CreditCardTransaction::NAME));
+
+        $oSession = $this->getSession();
+        $sSid = $oSession->sid(true);
+        if ($sSid != '') {
+            $sSid = '&' . $sSid;
+        }
+
+        $sModuleToken = PaymentGateway::getModuleToken($oSession);
+        $oTransaction->setTermUrl(
+            Registry::getConfig()->getCurrentShopUrl() . "index.php?cl=order&" . $sModuleToken . $sSid
+        );
+
+        $oTransactionService = $this->_getTransactionService();
+
+        $oPayment = oxNew(Payment::class);
+        $oPayment->load($oBasket->getPaymentId());
+
+        return "ModuleCreditCardForm.init(" .
+            $oTransactionService->getCreditCardUiWithData(
+                $oTransaction,
+                $this->_getPaymentAction($oPayment->oxpayments__wdoxidee_transactionaction->value),
+                Registry::getLang()->getLanguageAbbr()
+            ) . ")";
+    }
+
+    /**
+     * @return TransactionService
+     * @throws \Exception
+     */
+    private function _getTransactionService(): TransactionService
+    {
+        if (is_null($this->_oTransactionService)) {
+            $this->_oTransactionService = new TransactionService($this->_getCreditCardPaymentMethodConfig());
+        }
+
+        return $this->_oTransactionService;
+    }
+
+    /**
+     * @return CreditCardPaymentMethod
+     * @throws \Exception
+     */
+    private function _getCreditCardPaymentMethod(): CreditCardPaymentMethod
+    {
+        if (is_null($this->_oCreditCardPaymentMethod)) {
+            $this->_oCreditCardPaymentMethod = new CreditCardPaymentMethod();
+        }
+
+        return $this->_oCreditCardPaymentMethod;
+    }
+
+    /**
+     * @return Config
+     * @throws \Exception
+     */
+    private function _getCreditCardPaymentMethodConfig(): Config
+    {
+        if (is_null($this->_oPaymentMethodConfig)) {
+            $oPayment = $this->getPayment();
+            $this->_oPaymentMethodConfig = $this->_getCreditCardPaymentMethod()->getConfig($oPayment);
+        }
+
+        return $this->_oPaymentMethodConfig;
+    }
+
+    /**
+     * Converts the admin panel payment method action into a seamless
+     *L
+     *
+     * @param string $sAction
+     *
+     * @return string
+     */
+    private function _getPaymentAction(string $sAction): string
+    {
+        if ($sAction == 'pay') {
```

Comment:
please use paymentSDK constants for the purchase and authorization. Also i think there is one for pay

Reply:
👍